File: leetcode/TwoSum.cs
/* 
 * Given an array of integers nums and an integer target, return 
 * indices of the two numbers such that they add up to target.
 *
 * You may assume that each input would have exactly one solution,
 * and you may not use the same element twice.
 *
 * You can return the answer in any order.
 *
 * Follow-up: Can you come up with an algorithm that is less than O(n^2) time complexity?
 */
public class TwoSum {
    
    //O(n^2) solution
    public int[] TwoSum(int[] nums, int target) {
        int[] indexes = new int[2];
        bool isDone = false;

        for (int i = 0; i< nums.Length; i++){
            for (int j = i+1; j<nums.Length; j++){
                
                //skip the current index
                if (i == j){
                    continue;
                }
                
                //compare values to target
                if (nums[i]+nums[j] == target){
                    indexes[0] = i;
                    indexes[1] = j;
                    isDone = true;
                    break;
                }
            }
            if (isDone){ break; }
        }
        return indexes;
    }
}

File: leetcode/TwoSum.py
""" 
Given an array of integers nums and an integer target, return 
indices of the two numbers such that they add up to target.
 
You may assume that each input would have exactly one solution,
and you may not use the same element twice.
 
You can return the answer in any order.

Follow-up: Can you come up with an algorithm that is less than O(n^2) time complexity?
"""

class TwoSum(object):
    
    #O(n^2) solution
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        indexes = [0, 0]
        isDone = False

        for i in range(0, len(nums)):
            for j in range(i, len(nums)):

                if i==j:
                    continue
                
                if nums[i] + nums[j] == target:
                    indexes[0] = i
                    indexes[1] = j
                    isDone = True
                    break
            
            if isDone:
                break
        return indexes